src/utils/service_builder.rs

```diff
@@ -237,7 +237,8 @@ fn build_providers(
         feature = "pkcs11-provider",
         feature = "tpm-provider"
     )),
-    allow(unused_variables)
+    allow(unused_variables),
+    allow(clippy::match_single_binding)
```

Comment:
Increasing the `derivative` version does not help with this lint?

Reply:
no, mostly because `derivative` doesn't have anything to do with this code

Reply:
Isn't it [that one](https://github.com/mcarton/rust-derivative/issues/58)?

Reply:
It is, but that was something internal to derivative. This change is something that was happening in our code.

Reply:
Ah sorry I do not get it, is this Clippy lint getting triggered because of our code?
I increased the `derivative` version locally and executed:
```
cargo clippy --all-targets --features all-providers -- -D clippy::all -D clippy::cargo
```
and there were no errors?

Reply:
oh ok, got it when compiling with no features:
```rust
error: this match could be replaced by its body itself
   --> src/utils/service_builder.rs:243:5
    |
243 | /     match config {
244 | |         #[cfg(feature = "mbed-crypto-provider")]
245 | |         ProviderConfig::MbedCrypto { .. } => {
246 | |             info!("Creating a Mbed Crypto Provider.");
...   |
296 | |         }
297 | |     }
    | |_____^
    |
    = note: `-D clippy::match-single-binding` implied by `-D clippy::all`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding
help: consider using the match body instead
    |
243 |     {
244 |         error!(
245 |             "Provider \"{:?}\" chosen in the configuration was not compiled in Parsec binary.",
246 |             config
247 |         );
248 |         Err(Error::new(ErrorKind::InvalidData, "provider not compiled"))
  ...
```

Reply:
In that case I think it would be better to only allow the lint on that specific match as such:
```diff
diff --git a/src/utils/service_builder.rs b/src/utils/service_builder.rs
index bc2e228..825effc 100644
--- a/src/utils/service_builder.rs
+++ b/src/utils/service_builder.rs
@@ -240,6 +240,8 @@ fn build_providers(
     allow(unused_variables)
 )]
 unsafe fn get_provider(config: &ProviderConfig, key_id_manager: KeyIdManager) -> Result<Provider> {
+    // This lint will get triggered when no providers are included in the features.
+    #[allow(clippy::match_single_binding)]
     match config {
         #[cfg(feature = "mbed-crypto-provider")]
         ProviderConfig::MbedCrypto { .. } => {
```

Reply:
Oh you actually did that! I misread it sorry! In my head you were activating it for the whole crate